pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -1636,7 +1650,601 @@ var _ = Describe("Test Work Generator Controller", func() {
+				ObjectMeta: metav1.ObjectMeta{
+					Name: memberClusterName,
+				},
+			}
+			Expect(k8sClient.Create(ctx, &memberCluster)).Should(Succeed(), "Failed to create member cluster")
+
+			memberClusterNamespaceName = fmt.Sprintf(utils.NamespaceNameFormat, memberClusterName)
+			memberClusterReservedNS := corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: memberClusterNamespaceName,
+				},
+			}
+			Expect(k8sClient.Create(ctx, &memberClusterReservedNS)).Should(Succeed(), "Failed to create namespace")
+
+			testCRPName = "crp-" + utils.RandStr()
+		})
+
+		It("can create master resource snapshot and resource binding", func() {
+			// Create the master resource snapshot.
+			masterResourceSnapshot = generateResourceSnapshot(
+				1, 1, 0,
+				[][]byte{
+					testResourceCRD, testNameSpace, testResource,
+				})
+			Expect(k8sClient.Create(ctx, masterResourceSnapshot)).Should(Succeed(), "Failed to create master resource snapshot")
+
+			resourceBindingSpec := placementv1beta1.ResourceBindingSpec{
+				State:                placementv1beta1.BindingStateBound,
+				ResourceSnapshotName: masterResourceSnapshot.Name,
+				TargetCluster:        memberClusterName,
+			}
+			createClusterResourceBinding(&clusterResourceBinding, resourceBindingSpec)
+		})
+
+		It("should create work object with no apply strategy", func() {
+			workName := fmt.Sprintf(placementv1beta1.FirstWorkNameFmt, testCRPName)
+			applyStrategyUpdatedActual := workApplyStrategyUpdatedActual(memberClusterNamespaceName, workName, nil)
+			Eventually(applyStrategyUpdatedActual, timeout, interval).Should(Succeed(), "Failed to create expected work object")
+		})
+
+		It("can update apply strategy on the resource binding", func() {
+			// Update the apply strategy on the resource binding; use an Eventually
+			// block to avoid conflict induced errors.
+			Eventually(func() error {
+				if err := k8sClient.Get(ctx, types.NamespacedName{Name: clusterResourceBinding.Name}, clusterResourceBinding); err != nil {
+					return fmt.Errorf("failed to get resource binding: %w", err)
+				}
+
+				clusterResourceBinding.Spec.ApplyStrategy = &placementv1beta1.ApplyStrategy{
+					ComparisonOption: placementv1beta1.ComparisonOptionTypeFullComparison,
+					WhenToApply:      placementv1beta1.WhenToApplyTypeIfNotDrifted,
+					Type:             placementv1beta1.ApplyStrategyTypeServerSideApply,
+					WhenToTakeOver:   placementv1beta1.WhenToTakeOverTypeNever,
+				}
+
+				if err := k8sClient.Update(ctx, clusterResourceBinding); err != nil {
+					return fmt.Errorf("failed to update resource binding: %w", err)
+				}
+				return nil
+			}, timeout, interval).Should(Succeed(), "Failed to update apply strategy on the resource binding")
+		})
+
+		It("can add the rollout started condition to the resource binding", func() {
+			updateRolloutStartedGeneration(&clusterResourceBinding)
+		})
+
+		It("should update apply strategy on the work object", func() {
```

Comment:
why not continue to check the binding status too?

Reply:
Added the check 👌